tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/FindReplaceDialogTest.java

```diff
@@ -414,6 +415,18 @@ public void testActivateWholeWordsAndSearchForTwoWords() {
 		assertFalse(dialog.wholeWordCheckBox.getEnabled());
 	}
 
+	@Test
+	public void testIncrementalSearchOnlyActiveWhenAllowed() {
```

Comment:
The name of the method mentions "`active`" but it actually checks whether or not the checkbox is **enabled**. Either the name is wrong or the assertions.

Since I don't see any method in `FindReplaceDialog` that can give you the state of `findReplaceLogic.isIncrementalSearchAvailable()` for this test to check it, I assume a better name for the test would be:
```suggestion
	public void testIncrementalSearchOnlyEnabledWhenAllowed() {
```

Reply:
https://github.com/eclipse-platform/eclipse.platform.ui/commit/859fef9745d8b5af0d1c8d0a5874a5236329b0cd

Good catch! Although now this becomes inconsistent with `testActivateWholeWordsAndSearchForTwoWords` - I wonder if it makes sense to open a new PR to also change that one (or is it fine changing it in this PR?)

Reply:
I'd split that one into 2 (in another PR):
* `testWholeWordsOnlyEnabledWhenSearchingForOneWord`
* `testWholeWordsDisnabledWhenSearchingForRegEx`

The test does more than it says anyway, which is confusing